### ISSUE TITLE
Use u128 instead of bignums for STREAM

### ISSUE THREAD

**Issue by emschwartz** [emschwartz/interledger-rs#12]

**Comment by emschwartz** [comment]
Or maybe u64s instead

**Comment by tarcieri** [comment]
I'm not sure if things have changed, but in the past I've had trouble with rustc wanting nonexistent platform (i.e. `librt`) intrinsics for doing u128 operations which weren't available on 32-bit platforms, if you care about those.

That said, I *thought* LLVM could provide u128 on most platforms itself without `librt` dependencies, so maybe I'm recalling something from before `u128` was stabilized.

**Comment by ekrenzke** [comment]
Right now, BigUint is only being used as a convenient way to perform conversions to/from numbers of arbitrary size to a vector of `u8`s for oer. Now that `to_be_bytes(self) -> [u8; 8]` for `u64` has stabilized, this _should_ be a relatively simple fix.

**Comment by ekrenzke** [comment]
Still chugging along on this one. I ran into a brief issue with leading/trailing zeros, but I have identified test cases that will work towards improving it. I have also ported some of the test cases from the Javascript version, but I am obviously not certain on the validity since I do not know how they were generated. I am also looking into fuzzing in order to achieve parity across JS/RS implementations. I am happy to report that I have found a few bugs along the way -- more details will be provided later.
